bundles/org.openhab.binding.freeathomesystem/src/main/java/org/openhab/binding/freeathomesystem/internal/datamodel/FreeAtHomeDatapointGroup.java

```diff
@@ -0,0 +1,291 @@
+
+                    pairingId = neededPairingId;
+
+                    break;
+                }
+                case OUTPUT: {
+                    outputDatapoint = newDatapoint;
+
+                    if (inputDatapoint == null) {
+                        pairingId = neededPairingId;
+                    }
+
+                    break;
+                }
+            }
+        }
+
+        if (inputDatapoint != null && outputDatapoint != null) {
+            datapointGroupDirection = DatapointGroupDirection.INPUTOUTPUT;
+        } else {
+            if (inputDatapoint == null && outputDatapoint != null) {
+                datapointGroupDirection = DatapointGroupDirection.OUTPUT;
+            } else {
+                if (inputDatapoint != null && outputDatapoint == null) {
+                    datapointGroupDirection = DatapointGroupDirection.INPUT;
+                } else {
+                    datapointGroupDirection = DatapointGroupDirection.UNDEFINED;
+                }
+            }
+        }
+
+        return resultingDirection != DatapointDirection.UNKNOWN;
+    }
+
+    public void applyChangesForVirtualDevice() {
+        // The input and output datapoints are meant from the device point of view. Because the virtual devices are
+        // outside of the free@home system the input and output datapoint must be switched
+        @Nullable
+        FreeAtHomeDatapoint localDatapoint = inputDatapoint;
+
+        inputDatapoint = outputDatapoint;
+        outputDatapoint = localDatapoint;
+
+        if (inputDatapoint != null && outputDatapoint != null) {
+            datapointGroupDirection = DatapointGroupDirection.INPUTOUTPUT;
+        } else {
+            if (inputDatapoint == null && outputDatapoint != null) {
+                datapointGroupDirection = DatapointGroupDirection.OUTPUT;
+            } else {
+                if (inputDatapoint != null && outputDatapoint == null) {
+                    datapointGroupDirection = DatapointGroupDirection.INPUT;
+                } else {
+                    datapointGroupDirection = DatapointGroupDirection.UNDEFINED;
+                }
+            }
+        }
+
+        return;
+    }
+
+    public @Nullable FreeAtHomeDatapoint getInputDatapoint() {
+        return inputDatapoint;
+    }
+
+    public @Nullable FreeAtHomeDatapoint getOutputDatapoint() {
+        return outputDatapoint;
+    }
+
+    public DatapointGroupDirection getDirection() {
+        return datapointGroupDirection;
+    }
+
+    public boolean isDecimal() throws FreeAtHomeGeneralException {
+        boolean ret = false;
+
+        functionString = PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId));
+
+        switch (functionString) {
+            case PidTranslationUtils.PID_VALUETYPE_DECIMAL:
+                ret = true;
+                break;
+            default:
+                ret = false;
+                break;
+        }
+
+        return ret;
```

Comment:
Consider simplifying:
```suggestion
        return PidTranslationUtils.PID_VALUETYPE_DECIMAL.equals(PidTranslationUtils.getValueTypeForPairingId(String.format("0x%04X", pairingId)));
```

(same below)